client/config.c

```diff
@@ -440,3 +456,119 @@ _TDNFConfigReadPluginSettings(
 error:
     goto cleanup;
 }
+
+uint32_t
+TDNFReadMinVersionsFiles(
+    char *pszDir,
+    char ***pppszMinVersions
+    )
+{
+    uint32_t dwError = 0;
+    DIR *pDir = NULL;
+    struct dirent *pEnt = NULL;
+    char *pszFile = NULL;
+    char **ppszNewMinVersions = NULL;
+    char ***pppszArrayList = NULL;
+    int nFileCount = 0;
+    int i, j, k;
+    int nLineCount = 0;
+    int nTmp;
+
+    pDir = opendir(pszDir);
+    if (pDir == NULL)
+    {
+        goto cleanup;
+    }
+
+    while((pEnt = readdir(pDir)) != NULL)
+    {
+        if (fnmatch("*.conf", pEnt->d_name, 0) != 0)
+        {
+            continue;
+        }
+        nFileCount++;
+    }
+    closedir(pDir);
+
+    dwError = TDNFAllocateMemory(nFileCount + 1, sizeof(char **), (void **)&pppszArrayList);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    i = 0;
+    pDir = opendir(pszDir);
+    while((pEnt = readdir(pDir)) != NULL && i < nFileCount)
+    {
+        if (fnmatch("*.conf", pEnt->d_name, 0) != 0)
+        {
+            continue;
+        }
+        dwError = TDNFAllocateStringPrintf(&pszFile, "%s/%s", pszDir, pEnt->d_name);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        dwError = TDNFReadFileToStringArray(pszFile, &pppszArrayList[i]);
+        BAIL_ON_TDNF_ERROR(dwError);
+        i++;
```

Comment:
shouldn't we free pszFile here, as its allocated in loop.

Reply:
Good catch! Fixed.